extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLRecorder.java

```diff
@@ -0,0 +1,82 @@
+package io.quarkus.smallrye.graphql.runtime;
+
+import java.util.function.Supplier;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.runtime.ShutdownContext;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.smallrye.graphql.runtime.spi.QuarkusClassloadingService;
+import io.quarkus.vertx.http.runtime.ThreadLocalHandler;
+import io.smallrye.graphql.cdi.producer.GraphQLProducer;
+import io.smallrye.graphql.schema.model.Schema;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+/**
+ * Recorder
+ * 
+ * @author Phillip Kruger (phillip.kruger@redhat.com)
+ */
+@Recorder
+public class SmallRyeGraphQLRecorder {
+
+    public void createExecutionService(Schema schema) {
+        Arc.initialize(); // FIXME: why do I need this?
```

Comment:
This is incorrect. You probably want a `BeanContainerListener`.

Reply:
Thanks :)